### PR TITLE
Add type classification and originating document

### PR DESCRIPTION
**Pull request by TimvdLippe**
Work in progress for #281. Especially the classification  is really nasty, as we have to search through all initial keys and do a best-effort guess. An optimization I will do is to do a static lookup first for patterns that do not contain any variables.

See https://github.com/JetBrains/intellij-community/blob/master/platform/platform-resources-en/src/messages/InspectionsBundle.properties for an example of the patterns.

This PR does not include time-tracking yet, as I have yet to figure out how that will work out.

### CODE REVIEW

**Comment by TimvdLippe** [comment]
Reviewing is easier with ignoring the whitespace changes: https://github.com/TestRoots/watchdog/pull/282/files?w=1

File: intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/staticanalysis/CheckStyleStartup.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                .loadClass("com.puppycrawl.tools.checkstyle.PackageObjectFactory");
+
+        final Field packageObjectFactoryMapField = packageObjectFactoryClass.getDeclaredField("NAME_TO_FULL_MODULE_NAME");
+        packageObjectFactoryMapField.setAccessible(true);
+
+        return (Map<String, String>) packageObjectFactoryMapField.get(null);
+    }
+
+    // A bunch of reflection magic, as the plugin does not expose its internal classLoader
+    // TODO(timvdlippe): Update this code to use the type-safe methods once a PR to the plugin is merged that implements the method.
```
Comment (TimvdLippe) [inline intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/staticanalysis/CheckStyleStartup.java]: PR is open at https://github.com/jshiell/checkstyle-idea/pull/388

**Comment by Inventitech** [comment]
I think this no longer needs to be merged, right?

**Comment by TimvdLippe** [comment]
Yup fixed :tada:

**Pull request closed without merging**